packages/vxrn/src/plugins/clientBundleTreeShakePlugin.tsx

```diff
@@ -0,0 +1,99 @@
+import type { Node, Program } from 'estree'
+import { walk } from 'estree-walker'
+import MagicString from 'magic-string'
+import type { Plugin } from 'vite'
+
+interface TreeShakeTemplatePluginOptions {
+  sourcemap?: boolean
+}
+
+type AcornNode<N extends Node> = N & { start: number; end: number }
+
+export const clientBundleTreeShakePlugin = (options: TreeShakeTemplatePluginOptions) => {
+  return {
+    name: 'vxrn:server-tree-shake',
+    enforce: 'post',
+    transform(code, id) {
+      if (id.includes('node_modules')) {
+        return
+      }
+      if (!code.includes(`generateStaticParams`)) {
+        return
+      }
+
+      const s = new MagicString(code)
+      const codeAst = this.parse(code) as AcornNode<Program>
+
+      walk(codeAst, {
+        enter: (node) => {
+          if (node.type === 'ExportNamedDeclaration' || node.type === 'VariableDeclaration') {
+            let declarators = (
+              'declarations' in node
+                ? node.declarations
+                : 'declaration' in node
+                  ? [node.declaration]
+                  : []
+            ) as any[]
+
+            let shouldRemove = false
+
+            declarators.forEach((declarator) => {
+              if (
+                declarator.id.type === 'Identifier' &&
+                declarator.id.name === 'generateStaticParams'
+              ) {
+                shouldRemove = true
+              }
+            })
+
+            if (shouldRemove) {
+              // @ts-ignore
+              s.remove(node.start, node.end + 1)
+
+              if (node.type === 'ExportNamedDeclaration') {
+                // remove import declaration if it exists
+                // @ts-ignore
+                removeImportDeclaration(codeAst, node, s)
+              }
+            }
+          }
+        },
+      })
+
+      if (s.hasChanged()) {
+        return {
+          code: s.toString(),
+          map: options.sourcemap ? s.generateMap({ hires: true }) : undefined,
+        }
+      }
+    },
+  } satisfies Plugin
+}
+
+function removeImportDeclaration(
+  ast: Program,
+  importName: string,
+  magicString: MagicString
+): boolean {
```

Comment:
Not sure what this function does

Reply:
wdym? i did copy this from a nuxt plugin but didnt really check it thoroughly. it removes the imports from that function

Reply:
Ok, I assume it's for removing imports after we remove the params function. If yes, as far as i know, dead code elimination by esbuild/vite would do it automatically, so we might not even need to do it anyway. Not 100% sure though.